elastic/kibana
# [Fleet] Add support for "Edit Package Policy" extensions using latest version of a package

Resolves #114785

Ref #112831

## Summary

Allows fleet UI extensions to opt into targeting the latest version of a package during edits, rather than the version specified on the current package policy.

## How to Test

1. Create an outdated integration policy for the `synthetics` package by specifying a version in the "Add Integration" URL, e.g.

http://localhost:5601/{BASE_PATH}/app/fleet/integrations/synthetics-0.1.0/add-integration

![Screen Shot 2021-10-18 at 10 44 39 AM](https://user-images.githubusercontent.com/6766512/137754009-57f7c0dc-0746-4e28-9da3-8d7a07439025.png)

2. Navigate anywhere else in Fleet, or to the Synthetics integration's settings page and verify that the Synthetics package is auto-updated by referencing Kibana logs outputted during Fleet's "Ensure Preconfiguration" call

![Screen Shot 2021-10-18 at 10 46 57 AM](https://user-images.githubusercontent.com/6766512/137754384-6ab97d9f-e2a3-451c-923a-ba180cf4d4e2.png)

3. Navigate back to the policies list for Synthetics, and click the integration policy name to navigate to the "Edit Integration" page

![Screen Shot 2021-10-18 at 10 48 47 AM](https://user-images.githubusercontent.com/6766512/137755065-6fa5273b-410e-46e0-9b5e-f4ec6f0617f7.png)

4. Note that the UI displays "Upgrade" content instead of "Edit", indicating that the package policy will be upgraded

![Screen Shot 2021-10-18 at 10 50 57 AM](https://user-images.githubusercontent.com/6766512/137755293-7f32af07-9190-40e6-b927-1cd8965c38dc.png)

5. Address any conflicts and save

6. Note the policy has been upgraded to `0.3.0`
